packages/vuetify/src/components/VSwitch/VSwitch.sass

```diff
@@ -1,132 +1,137 @@
-@import './_variables.scss'
-
-// Theme
-+theme(v-input--switch) using ($material)
-  .v-input--switch__thumb
-    color: map-deep-get($material, 'selection-controls', 'thumb', 'inactive')
-
-  .v-input--switch__track
-    color: map-deep-get($material, 'selection-controls', 'track', 'inactive')
-
-  &.v-input--is-disabled:not(.v-input--is-dirty)
-    .v-input--switch__thumb
-      color: map-deep-get($material, 'selection-controls', 'thumb', 'disabled') !important
-
-    .v-input--switch__track
-      color: map-deep-get($material, 'selection-controls', 'track', 'disabled') !important
-
-.v-input--switch
-  &__track,
-  &__thumb
-    background-color: currentColor
-    pointer-events: none
-    transition: inherit
-
-  &__track
-    border-radius: $switch-track-border-radius
-    width: $switch-track-width
-    height: $switch-track-height
-    left: $switch-track-x
-    position: absolute
-    opacity: $switch-track-opacity
-    right: $switch-track-x
-    top: $switch-track-top
-
-  &__thumb
+@forward './variables'
+@use '../../styles/settings'
+@use '../../styles/tools'
+@use './variables' as *
+
+.v-switch
+  .v-label
+    margin-inline-start: $switch-label-margin-inline-start
+
+.v-switch__track,
+.v-switch__thumb
+  background-color: currentColor
+  transition: none
+
+  .v-selection-control--error &
+    background-color: $switch-error-background-color
+
+.v-switch__track
+  border-radius: $switch-track-border-radius
+  height: $switch-track-height
+  left: $switch-track-x
+  opacity: $switch-track-opacity
+  pointer-events: none
+  position: absolute
+  right: $switch-track-x
+  top: 50%
+  transform: translateY(-50%)
+  width: $switch-track-width
```

Comment:
This should be `--v-selection-control-size - $switch-track-x * 2`

Reply:
Need your help with this one, not understanding what this changes.